app/main.py

```diff
@@ -1,6 +1,19 @@
-from typing import Callable
+from typing import Any, Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    @wraps(func)
+    def inner(*args) -> Any:
+        temp = tuple(args)
```

Comment:
You should check both `args` and `kwargs` because the function might accept the same `args` but different `kwargs`. Additionally, I recommend checking the function name since there could be two different functions that accept the same arguments.